segmentstore/server/src/main/java/io/pravega/segmentstore/server/containers/MetadataCleaner.java

```diff
@@ -179,6 +180,26 @@ protected void doStop() {
                 }, this.executor);
     }
 
+    private CompletableFuture<Void> deleteUnusedTransientSegments() {
+        long lastSeqNo = this.lastIterationSequenceNumber.getAndSet(this.metadata.getOperationSequenceNumber());
+        long traceId = LoggerHelpers.traceEnterWithContext(log, this.traceObjectId, "deleteUnusedTransientSegments", lastSeqNo);
+
+        // Serialize only those segments that are still alive (not deleted or merged - those will get removed anyway).
+        this.metadata.getEvictionCandidates(lastSeqNo, this.config.getMaxConcurrentSegmentEvictionCount())
+                .stream()
+                .filter(sm -> sm.getType() == SegmentType.TRANSIENT_SEGMENT && sm.isDeleted())
```

Comment:
You're getting all the deleted transient segments. Your code is not complete yet, but I assume you'll want to delete those, right? 

In that case, your filter should be `!sm.isDeleted()`.

Reply:
Sorry yes, thats right.